### PR TITLE
Support Op Goerli L2 and Op Mainnet L2

### PR DESCRIPTION
**Pull request by boyuan-chen**
- Add Op Mainnet and Goerli L2 configurations
- Fix RLP encoding for the deposit transaction's receipt
- Load EIP1559 parameters from the network configuration

### CODE REVIEW

File: consensus/misc/eip1559.go [hunk 1]
```diff
@@ -99,12 +99,12 @@ func CalcBaseFee(config *chain.Config, parent *types.Header) *big.Int {
 	}
 }
 
-func getBaseFeeChangeDenominator(borConfig *chain.BorConfig, number uint64) uint64 {
+func getBaseFeeChangeDenominator(config *chain.Config, number uint64) uint64 {
```
Comment (jyellick) [inline consensus/misc/eip1559.go:102]: Not that it's wrong, but I don't see why we're changing the Bor aspect of this? The only logical change seems to be the fall through return value?

**Review by jyellick — COMMENTED** [review]
Overall LGTM

**Review by mmontour1306 — APPROVED** [review]

File: consensus/misc/eip1559.go (hunk 1 above)
Comment (boyuan-chen) [reply consensus/misc/eip1559.go:102]: To get the denominator, we need `config` in the following line.
```
return config.BaseFeeChangeDenominator(params.BaseFeeChangeDenominator)
```

**Review by jyellick — APPROVED** [review]
LGTM

**Pull request merged**